src/tokenizer.ts

```diff
@@ -1528,7 +1528,18 @@ export class Tokenizer extends DiagnosticEmitter {
     var sepCount = this.readDecimalFloatPartial(false);
     if (this.pos < end && text.charCodeAt(this.pos) == CharCode.DOT) {
       ++this.pos;
-      sepCount += this.readDecimalFloatPartial();
+      if (this.pos < end && text.charCodeAt(this.pos) == CharCode.DOT) {
+        // try recognize potentially valid `123..identifier`
```

Comment:
Isn't this not part of the literal, but a property access on, say, `1.`? Why does this need to be handled in `readDecimalFloat`?

Reply:
This is necessary to properly take out errors for such cases:
```ts
1..1
```
but at the same time fast skip possible valid cases like:
```ts
1..toString()
```

Reply:
Generally this check is optional, without it we will still get an error message, but it will look like this: "Expression expected." instead "Identifier expected."

Reply:
Hmm, interesting. In V8, the error is

```
1..1
  ^^

Uncaught SyntaxError: Unexpected number
```

Reply:
Yeah. We also could use this error. But TS doesn't handle this properly and just split number to `1.` and `.1` and produce "';' expected." error

Reply:
Looks like V8's parser evaluates the kind of in-line starting token, regardless of the space:

```
1.. 1
    ^

Uncaught SyntaxError: Unexpected number
```

Reply:
You should remember we have problem with such cases when statements separated by whitespace

Reply:
Yeah, I remember. Is there maybe a way to fix this problem instead, say by adding a whitespace check when attempting to parse the next expression statement?

Reply:
It's a little different scope. I think it could somehow be added here: https://github.com/AssemblyScript/assemblyscript/blob/main/src/parser.ts#L4229

Which try to recover parsing after failed attempt. But I'm not sure